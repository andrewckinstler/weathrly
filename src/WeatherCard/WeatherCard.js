import React from 'react';
import moment from 'moment';
import './WeatherCard.scss'


export const WeatherCard = props => {
  const { date, weather, main } = props
  const icon = weather[0].icon
  const min = main.temp_min.toFixed(0);
  const max = main.temp_max.toFixed(0)

  return (
    <article className='card'>
      <h3>{moment(date).format('dddd l')}</h3>
      <img src={`http://openweathermap.org/img/wn/${icon}@2x.png`}  alt='condition icon'/>
      <p><span className='bold'>H: </span>{min}°/ <span className='bold'>L: </span>{max}°</p>
    </article>
  )
}