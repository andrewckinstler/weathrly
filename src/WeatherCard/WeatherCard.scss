.card {
  padding: 15px;
  border-radius: 10px;
  background-color: #fff;
  text-align: center;

  img { 
    padding: 7px 0;
  }
  
  .bold {
    font-weight: bold;
  }
}