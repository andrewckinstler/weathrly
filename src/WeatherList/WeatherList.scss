.card-container {
  display: flex;
  justify-content: space-around;
  padding: 35px 75px 150px 75px;
}