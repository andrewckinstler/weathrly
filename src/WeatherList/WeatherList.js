import React from 'react';
import './WeatherList.scss';
import { WeatherCard } from '../WeatherCard/WeatherCard'

export const WeatherList = ( props ) => {
  const createCards = props.props.map(obj => 
  <WeatherCard 
  key={obj.dt}
  date={obj.dt_txt}
  weather={obj.weather}
  main={obj.main}
  />)
  return (
    <div className='card-container'>{createCards}</div>
  )
}