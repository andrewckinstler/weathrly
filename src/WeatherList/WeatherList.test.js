import React from 'react';
import { shallow } from 'enzyme';
import { WeatherList } from './WeatherList';
import WeatherCard from '../WeatherCard/WeatherCard';

describe('Weather list', () => {
  it('should match a snapshot', () => {
    const mockForcast = [
      {
          "dt": 1580428800,
          "main": {
              "temp": 20.79,
              "feels_like": 13.06,
              "temp_min": 20.05,
              "temp_max": 20.79,
              "pressure": 1021,
              "sea_level": 1021,
              "grnd_level": 726,
              "humidity": 87,
              "temp_kf": 0.41
          },
          "weather": [
              {
                  "id": 600,
                  "main": "Snow",
                  "description": "light snow",
                  "icon": "13d"
              }
          ],
          "clouds": {
              "all": 96
          },
          "wind": {
              "speed": 4.45,
              "deg": 18
          },
          "snow": {
              "3h": 0.81
          },
          "sys": {
              "pod": "d"
          },
          "dt_txt": "2020-01-31 00:00:00"
      },
      {
          "dt": 1580439600,
          "main": {
              "temp": 11.8,
              "feels_like": 2.91,
              "temp_min": 11.25,
              "temp_max": 11.8,
              "pressure": 1028,
              "sea_level": 1028,
              "grnd_level": 728,
              "humidity": 84,
              "temp_kf": 0.31
          },
          "weather": [
              {
                  "id": 600,
                  "main": "Snow",
                  "description": "light snow",
                  "icon": "13n"
              }
          ],
          "clouds": {
              "all": 91
          },
          "wind": {
              "speed": 5.3,
              "deg": 314
          },
          "snow": {
              "3h": 0.31
          },
          "sys": {
              "pod": "n"
          },
          "dt_txt": "2020-01-31 03:00:00"
      },
      {
          "dt": 1580450400,
          "main": {
              "temp": 4.24,
              "feels_like": -4.95,
              "temp_min": 3.88,
              "temp_max": 4.24,
              "pressure": 1032,
              "sea_level": 1032,
              "grnd_level": 729,
              "humidity": 76,
              "temp_kf": 0.2
          },
          "weather": [
              {
                  "id": 803,
                  "main": "Clouds",
                  "description": "broken clouds",
                  "icon": "04n"
              }
          ],
          "clouds": {
              "all": 72
          },
          "wind": {
              "speed": 5.01,
              "deg": 300
          },
          "sys": {
              "pod": "n"
          },
          "dt_txt": "2020-01-31 06:00:00"
      }
    ];

    const wrapper = shallow(<WeatherList props={mockForcast} />)

    expect(wrapper).toMatchSnapshot()
  })
})