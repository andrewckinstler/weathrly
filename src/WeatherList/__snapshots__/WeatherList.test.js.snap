// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather list should match a snapshot 1`] = `
<div
  className="card-container"
>
  <WeatherCard
    date="2020-01-31 00:00:00"
    key="1580428800"
    main={
      Object {
        "feels_like": 13.06,
        "grnd_level": 726,
        "humidity": 87,
        "pressure": 1021,
        "sea_level": 1021,
        "temp": 20.79,
        "temp_kf": 0.41,
        "temp_max": 20.79,
        "temp_min": 20.05,
      }
    }
    weather={
      Array [
        Object {
          "description": "light snow",
          "icon": "13d",
          "id": 600,
          "main": "Snow",
        },
      ]
    }
  />
  <WeatherCard
    date="2020-01-31 03:00:00"
    key="1580439600"
    main={
      Object {
        "feels_like": 2.91,
        "grnd_level": 728,
        "humidity": 84,
        "pressure": 1028,
        "sea_level": 1028,
        "temp": 11.8,
        "temp_kf": 0.31,
        "temp_max": 11.8,
        "temp_min": 11.25,
      }
    }
    weather={
      Array [
        Object {
          "description": "light snow",
          "icon": "13n",
          "id": 600,
          "main": "Snow",
        },
      ]
    }
  />
  <WeatherCard
    date="2020-01-31 06:00:00"
    key="1580450400"
    main={
      Object {
        "feels_like": -4.95,
        "grnd_level": 729,
        "humidity": 76,
        "pressure": 1032,
        "sea_level": 1032,
        "temp": 4.24,
        "temp_kf": 0.2,
        "temp_max": 4.24,
        "temp_min": 3.88,
      }
    }
    weather={
      Array [
        Object {
          "description": "broken clouds",
          "icon": "04n",
          "id": 803,
          "main": "Clouds",
        },
      ]
    }
  />
</div>
`;
